Packs/ArmisEventCollector/Integrations/ArmisEventCollector/ArmisEventCollector_test.py

```diff
@@ -0,0 +1,393 @@
+            'events_last_fetch_time': '2023-01-01T01:00:30.123456+00:00', 'access_token': 'test_access_token'},
+        fetch_start_time,
+        ['Events'],
+        events_with_different_time_2,
+        events_with_different_time_2,
+        {'events_last_fetch_ids': ['4', '5', '6'],
+            'events_last_fetch_time': '2023-01-01T01:01:00.123456+00:00', 'access_token': 'test_access_token'}
+    )
+    case_second_fetch_with_duplicates = (  # type: ignore
+        1000,
+        {'events_last_fetch_ids': ['1', '2', '3'],
+            'events_last_fetch_time': '2023-01-01T01:00:30.123456+00:00', 'access_token': 'test_access_token'},
+        fetch_start_time,
+        ['Events'],
+        events_with_duplicated_from_1,
+        [{
+            'unique_id': '6',
+            'time': '2023-01-01T01:01:00.123456+00:00'
+        }],
+        {'events_last_fetch_ids': ['6'],
+            'events_last_fetch_time': '2023-01-01T01:01:00.123456+00:00', 'access_token': 'test_access_token'}
+    )
+
+    case_no_new_event_from_fetch = (  # type: ignore
+        1000,
+        {'events_last_fetch_ids': ['1', '2', '3'],
+            'events_last_fetch_time': '2023-01-01T01:00:30.123456+00:00', 'access_token': 'test_access_token'},
+        fetch_start_time,
+        ['Events'],
+        [],
+        [],
+        {'events_last_fetch_ids': ['1', '2', '3'],
+            'events_last_fetch_time': '2023-01-01T01:00:30.123456+00:00', 'access_token': 'test_access_token'}
+    )
+
+    case_all_events_from_fetch_have_the_same_time = (  # type: ignore
+        1000,
+        {'events_last_fetch_ids': ['1', '2', '3'],
+            'events_last_fetch_time': '2023-01-01T01:00:30.123456+00:00', 'access_token': 'test_access_token'},
+        fetch_start_time,
+        ['Events'],
+        events_with_same_time,
+        events_with_same_time,
+        {'events_last_fetch_ids': ['1', '2', '3', '4', '5', '6'],
+            'events_last_fetch_time': '2023-01-01T01:00:30.123456+00:00', 'access_token': 'test_access_token'}
+    )
+
+    @ pytest.mark.parametrize('max_fetch, last_run, fetch_start_time, event_types_to_fetch, response, events, next_run', [
+        case_first_fetch, case_second_fetch, case_second_fetch_with_duplicates,
+        case_no_new_event_from_fetch, case_all_events_from_fetch_have_the_same_time
+    ])
+    def test_fetch_flow_cases(self, mocker, dummy_client, max_fetch, last_run,
+                              fetch_start_time, event_types_to_fetch, response, events, next_run):
+        """
+        Given:
+            - Case 1: First fetch, response has 3 events with different timestamps.
+            - Case 2: Second fetch, response has 3 events with different timestamps.
+            - Case 3: Second fetch with duplicated, some events in the response have same timestamps as last fetch.
+            - Case 4: Second fetch with empty response (no new events from last fetch).
+            - Case 5: Second fetch, all events have the same timestamp.
```

Comment:
Lets add a case where we are in second fetch but the response had events in the same time as last fetch (dups between fetches)

Reply:
This case is tested in Case 3: Second fetch with duplicated, some events in the response have same timestamps as last fetch.
Am I missing something?